test/collector.test.ts

```diff
@@ -0,0 +1,351 @@
+import { expect, use } from 'chai';
+import Collector from '../build/contracts/Collector.json';
+import TestToken from '../build/contracts/TestToken.json';
+import {
+    deployContract,
+    loadFixture,
+    MockProvider,
+    solidity
+} from 'ethereum-waffle';
+import { Wallet } from '@ethersproject/wallet';
+
+use(solidity);
+
+const FIRST_PARTNER_SHARE = 20;
+const SECOND_PARTNER_SHARE = 30;
+const THIRD_PARTNER_SHARE = 40;
+const FOURTH_PARTNER_SHARE = 10;
+
+const buildPartners = (partners: Wallet[], shares?: number[]) => {
+    return [
+        {
+            beneficiary: partners[0].getAddress(),
+            share: shares?.[0] ?? FIRST_PARTNER_SHARE
+        },
+        {
+            beneficiary: partners[1].getAddress(),
+            share: shares?.[1] ?? SECOND_PARTNER_SHARE
+        },
+        {
+            beneficiary: partners[2].getAddress(),
+            share: shares?.[2] ?? THIRD_PARTNER_SHARE
+        },
+        {
+            beneficiary: partners[3].getAddress(),
+            share: shares?.[3] ?? FOURTH_PARTNER_SHARE
+        }
```

Comment:
surely this could be iterated...

Reply:
I implemented the iterated version but it was less readable since it implies the use of an array of cons and an extra variable for the size of the array. Those variables where used on a test bellow that was also less readable. I rolled it back to this version.

Reply:
would it? consider following:
```
const PARNER_SHARES = [20, 30, 40, 10];

type Partner = {
    beneficiary: string,
    share: number,
}

const buildPartners = (wallets: Wallet[], shares: number[]): Partner[] => {
    return wallets.map(({getAddress}: Wallet, partnerIndex) => ({
        beneficiary: getAddress(),
        share: shares?.[partnerIndex] ?? FIRST_PARTNER_SHARE
    }));
```

on the other hand, if you are manually defining these you might as well manually define the whole object without the need for this function in the first place ;)

Reply:
This is a very complex solution for that simple problem. It's more efficient, advanced and elegant but think in simplicity as what can be more understandable for a junior developer. 
The code you propose here is not working for me by the way. I did something in between.

Reply:
then, as I said above, just hardcode the entire object, no need for functions. Btw:
> It's more efficient, advanced, elegant and shows a lot of knowledge in Typescript but think of simplicity as what can be more understandable for a junior developer
is not a good argument.
1. if something has those properties you layed out, it IS more understandable to junior dev
2. it does not show a lot of knowledge in TS at all. the only ts thing here is one type
3. my suggestion IS more simple
4. even junior developer would be expected to write clean code
5. tests need to follow clean code principals, as they are the single source of truth about the code without looking at the code

Reply:
Done. Regarding this we should define a process for the cases where reviewer and owner disagree. For this case I just want to close this PR so I implemented an even more complex version of the proposed solution but I don't think it's the best one.